lib/plug/static.ex

```diff
@@ -182,6 +186,100 @@ defmodule Plug.Static do
     h in only or match?({0, _}, prefix != [] and :binary.match(h, prefix))
   end
 
+  defp serve_range(conn, path, segments, [], %{gzip?: gzip?, brotli?: brotli?} = options) do
+    encoding = file_encoding(conn, path, gzip?, brotli?)
+    serve_static(encoding, segments, options)
+  end
+  defp serve_range(conn, path, segments, [range], options) do
+    encoding = file_encoding(conn, path, false, false)
+    serve_range(encoding, range, segments, options)
+  end
+  defp serve_range(_conn, _path, _segments, _range, _options) do
+    raise InvalidRangeError
+  end
+
+  @byte_range_pattern ~r/^\s*bytes=([0-9]+)?-([0-9]+)?\s*$/
+
+  defp serve_range({:ok, conn, file_info, path}, range, segments, options) do
+    file_size = elem(file_info, 1)
+
+    parsed_range =
+      @byte_range_pattern
+      |> Regex.run(range)
+      |> start_and_end(file_size)
+      |> check_bounds(file_size)
+
+    case parsed_range do
+      {range_start, range_end} ->
+        send_range(conn, path, range_start, range_end, file_size, segments, options)
+      :entire_file ->
+        serve_static({:ok, conn, file_info, path}, segments, options)
+      :invalid ->
+        raise InvalidRangeError
+    end
+  end
+
+  defp serve_range({:error, conn}, _range, _segments, _options) do
+    conn
+  end
+
+  defp start_and_end([_, "", ""], _file_size), do: :invalid
+
+  defp start_and_end([_, range_start], file_size)
+    when is_binary(range_start)
+  do
+    {to_integer(range_start), file_size - 1}
+  end
+
+  defp start_and_end([_, "", tail_length], file_size)
+    when is_binary(tail_length)
+  do
+    {file_size - to_integer(tail_length), file_size - 1}
+  end
+
+  defp start_and_end([_, range_start, range_end], _file_size) do
+    {to_integer(range_start), to_integer(range_end)}
+  end
+
+  defp start_and_end(_range, _file_size), do: :invalid
+
+  defp check_bounds(:invalid, _file_size), do: :invalid
+
+  defp check_bounds({range_start, range_end}, file_size)
+    when range_start < 0 or range_end >= file_size or range_start > range_end
+  do
+    :invalid
+  end
+
+  defp check_bounds({0, range_end}, file_size)
+    when range_end == file_size - 1
+  do
+    :entire_file
+  end
+
+  defp check_bounds({range_start, range_end}, _file_size) do
+    {range_start, range_end}
+  end
+
+  defp to_integer(str) do
+    {num, _} = Integer.parse(str)
+    num
+  end
+
+  defp send_range(conn, path, range_start, range_end, file_size, segments, options) do
+    %{headers: headers} = options
+    length = (range_end - range_start) + 1
+    content_type = segments |> List.last |> MIME.from_path
```

Comment:
You need to support custom content types, as here: https://github.com/scouten/plug/blob/a20a9a45aa04e86291ef6e67cd6b5390ccfba8ce/lib/plug/static.ex#L288-L289